File: js/nova/network/ClientSocket.js
var ClientSocket = function () {
	var self = this;
	
	this.socket = nova_null;
	this.connection = nova_null;
	
	this.ClientSocket = function () {
		self.this();
	};
	
	this.destroy = function () {
	};
	
	this.this = function () {
	};
	
	this.connect = function (ipAddress, port) {
		self.socket = self.nova_clientsocket_connect(ipAddress.chars.data, port);
		if (self.socket == 0) {
			return false;
		}
		self.connection = new ConnectionSocket(self.socket);
		return true;
	};
	
	this.close = function () {
		var result;
		result = self.nova_socket_close(self.socket);
		if (result == 0) {
			return false;
		}
		return true;
	};
	
	this.super = function () {
	};
	
	
};

File: java/nova/network/ClientSocket.java
package nova.network;

import java.util.Optional;
import nova.exception.ExceptionData;
import nova.exception.Exception;
import nova.exception.DivideByZeroException;
import nova.io.NovaConsole;
import nova.primitive.number.Number;
import nova.primitive.number.NovaByte;
import nova.primitive.number.NovaShort;
import nova.primitive.number.NovaInt;
import nova.primitive.number.NovaLong;
import nova.primitive.number.NovaFloat;
import nova.primitive.number.NovaDouble;
import nova.primitive.Null;
import nova.primitive.number.Char;
import nova.primitive.Bool;
import nova.datastruct.list.NovaArray;
import nova.datastruct.list.IntArray;
import nova.datastruct.list.CharArray;
import nova.datastruct.list.DoubleArray;
import nova.datastruct.list.IntRange;
import nova.thread.Thread;
import nova.thread.async.Async;
import nova.gc.GC;
import nova.math.NovaMath;
import nova.NovaObject;
import nova.NovaString;
import nova.System;
import nova.Class;
import nova.network.ConnectionSocket;
import nova.network.Socket;

public class ClientSocket extends Socket
{
	private SOCKET_ID_TYPE socket;
	
	public ConnectionSocket connection;
	
	
	
	
	public ClientSocket()
	{
		init();
	}
	
	
	
	public void init()
	{
	}
	
	public boolean connect(NovaString ipAddress, int port)
	{
		socket = nova_clientsocket_connect(ipAddress.chars.data, port);
		if (socket == 0)
		{
			return false;
		}
		connection = new ConnectionSocket(socket);
		return true;
	}
	
	public boolean close()
	{
		int result;
		result = nova_socket_close(socket);
		if (result == 0)
		{
			return false;
		}
		return true;
	}
	
}
